### DISCUSSION TITLE
Ideas / Feature Discussion

### DISCUSSION THREAD

**Discussion by KaneFreeman** [StaticJsCMS/static-cms#4] [Ideas]
## Feature Discussion

While the initial work will be focused on modernizing the code base as well as updating the documentation to reflect the changes, new ideas are always welcome! Please add any ideas you have for future features below.

## Roadmap

You can see the current plans in the [Roadmap](#2)

**Comment by pensivedog** [comment]
Thanks for working on this! I'm really looking forward to transitioning from Netlify CMS to StaticCMS once a stable release is ready.

The main question I have is regarding the project management and future development. How can this project be set up to avoid a similar eventual fate as Netlify CMS? In general, how can projects with broad adoption and public benefit avoid being so dependent on a single person?

**Reply by KaneFreeman** [reply]
I think the most important thing will be establishing the governance model for the project, something I am still working on figuring out the details for. After that it getting contributors for the project.

Eventually if the project gets enough momentum I would probably look into having the project join a foundation (like https://openjsf.org/ for example).

**Comment by ryangittings** [comment]
It would be great to edit slugs/filenames within the CMS! Probably the most requested thing from my clients.

**Comment by robigan** [comment]
Can we have a List widget that's just an array of strings, because right now for the List component, it has to be an array of objects like this
<img width="292" alt="image" src="https://user-images.githubusercontent.com/35210888/204331754-765abfd8-e874-4540-919f-0e186e019249.png">
but for the tags on my blog I just want an array of strings
<img width="109" alt="image" src="https://user-images.githubusercontent.com/35210888/204331881-84c0e4c7-c8d2-42b8-bdd3-6e1aba819f6e.png">

**Reply by robigan** [reply]
Unless it's already possible but I can't figure it out

**Comment by denyskon** [comment]
For some use cases, I have long list fields. When adding many entries, you always have to scroll back to the top to reach the add button. It would be great if we could get the possibility to move the add button to the bottom of a list widget.

**Comment by denyskon** [comment]
It would be really great if we could get an icon picker widget for Material Design Icons and/or Font Awesome. On my site I have many sections where I have to pick an icon, and for now I have to manually search for the icon and enter its name.

**Reply by mnordhaus** [reply]
I thought about that one as well. Fortunately staticcms allows to create your own widgets. I am not committing yet to implement that as I have other things I would like to provide to the project. But gathering ideas on how it could look like might not harm. Maybe putting some issue together or a separate project? Like a "StaticCMS Widget Extension Pack"? I have some ideas for more widgets.

**Reply by KaneFreeman** [reply]
I'll make a dedicated repo for more specialized widgets, and probably move several of the existing (but less "core") widgets over to it (map, key/value, uuid, etc). I am planning on making this a big part of the `v5` release later this year, to keep the breaking changes in a major release.

Once the repo is setup and the widgets moved, I will deprecate the ones in the core repo.

Then we can start suggesting new widgets in the new repo.

**Reply by KaneFreeman** [reply]
I made an issue for this #1048
